client/src/components/ReviewsList.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import ReviewsListEntity from './ReviewsListEntity.jsx';
+import ReviewsPages from './ReviewsPages.jsx';
+
+class ReviewsList extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+        ordered_reviews: this.props.reviews,
+        rendering_order: 'Most recent',
+        page_num: 1
+    }
+
+    this.changeReviewsOrder = this.changeReviewsOrder.bind(this);
+    this.changeReviewsPage = this.changeReviewsPage.bind(this);
+  }
+
+  changeReviewsOrder(event) {
+      // sort reviews according to selected order
+      let order = event.target.value;
+      let ordered_reviews = [];
+
+      if (order === 'Most recent') {
+        ordered_reviews = this.props.reviews.sort( (a,b) => {
+            return new Date(b.update_date) - new Date(a.update_date);
+        });
+
+      } else if (order === 'Highest rated') {
+        ordered_reviews = this.props.reviews.sort( (a,b) => {
+            return b.rating - a.rating;
+        });
+
+      }
+
+      this.setState({
+          ordered_reviews: ordered_reviews,
+          rendering_order: order,
+          page_num: 1
+      });
+  }
+
+  changeReviewsPage(event) {
+
+      let value = parseInt(event.target.className);
+      if (value === -1) {
+        // user clicked prev if page number is less than 1 keep it 1
+        if (this.state.page_num > 1) {
+            value = this.state.page_num - 1;
+        } else {
```

Comment:
This `else` block isn't needed because it will reach the end of the function anyway. If `value === -1`, then it won't go into the `else if` block with `value === 0`. So it will just go to the end of the function. You can save a few lines here and in the `value === 0` block too.

Reply:
Actually never mind, I saw that you are use `setState` at the end.